src/location-picker/location-picker.service.ts

```diff
@@ -2,34 +2,80 @@ import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 import { Observable } from 'rxjs/Observable';
 
-import { LocationPickerValue } from './location-picker.types';
+import {
+  LocationPickerValue,
+  LatLngCoordinate,
+} from './location-picker.types';
 
 @Injectable()
 export class LocationPickerService {
-    constructor(
-        private http: HttpClient
+  constructor(
+    private http: HttpClient
     ) {}
 
-    public getLocationsByQuery(
-        /**
-         * The URL for contacting the BFF,
-         * appending search=<search> as query argument.
-         */
-        dataSource: string,
-        /** The string to search for */
-        search: string,
-        /** The types of locations to search for (comma-separated) */
-        types: string
-    ): Observable<LocationPickerValue[]> {
-        if (typeof dataSource === 'string') {
-            const uri = dataSource +
-                ((dataSource.indexOf('?') < 0) ? '?' : '&') +
-                'search=' + search +
-                (types ? '&types=' + types : '');
-            return this.http.get<LocationPickerValue[]>(uri);
-        } else {
-            // should never happen
-            throw new TypeError('unsupported dataSource type "' + (typeof dataSource) + '"');
-        }
+  public getLocationsByQuery(
+    /**
+     * The URL for contacting the BFF,
+     * appending search=<search> as query argument.
+     */
+     dataSource: string,
+     /** The string to search for */
+     search: string,
+     /** The types of locations to search for (comma-separated) */
+     types: string
+     ): Observable<LocationPickerValue[]> {
+    if (typeof dataSource === 'string') {
+      const dataParts = this.analyseDataSource(dataSource);
+      const uri = dataParts.uri +
+        '/locations' +
+        dataParts.params +
+        'search=' + search +
+        (types ? '&types=' + types : '');
+      return this.http.get<LocationPickerValue[]>(uri);
+    } else {
+      // should never happen
+      throw new TypeError('unsupported dataSource type "' + (typeof dataSource) + '"');
     }
+  }
+
+  public getLocationByCoordinates(
+    /**
+     * The URL for contacting the BFF,
+     * appending search=<search> as query argument.
+     */
+     dataSource: string,
+     /** The latitude and longitude coordinates to searxh for */
+     latLong: LatLngCoordinate
+     ): Observable<any> {
+    if (typeof dataSource === 'string') {
+      const dataParts = this.analyseDataSource(dataSource);
+      const uri = dataParts.uri +
+        '/coordinates' +
+        dataParts.params +
+        'lat=' + latLong.lat +
+        '&lng=' + latLong.lng;
+      return this.http.get<LocationPickerValue[]>(uri);
+    } else {
+      // should never happen
+      throw new TypeError('unsupported dataSource type "' + (typeof dataSource) + '"');
+    }
+  }
+
+  private analyseDataSource(dataSource: string) {

```

Comment:
I've made small changes to this code to make it more readable 
(your  code will also work, you deside to replace it or not).

-  Single letter vars don't describe there contents  
- I prefer no else statement if unnecessary 
```javascript
    if (dataSource.indexOf('?') === -1) {
      return { uri: dataSource, params: '?'};
    }

    const dataSourceParts = dataSource.split('?');
    const uri = dataSourceParts[0];
    const params = `?${dataSourceParts[1]}&`;

    return { uri, params };
```